ChangeLog.d/add-sm3.txt

```diff
@@ -0,0 +1,2 @@
+Features
```

Comment:
Regarding the commit structure: the initial commit is too large. Please split it into meaningful parts. For example: creation of `sm3.c` and `sm3.h` and adding them to build scripts; declaration of the new config options; adding the sm3 test suite; adding md dispatch code; adding md tests; adding psa dispatch code; adding psa tests.

After the first round of review, please make any requested changes through subsequent commits. Avoid rebasing unless it's to resolve a merge conflict, and if you do rebase, please keep the same merge structure.

Reply:
When you fix things after a review, please write meaningful commit messages and don't group unrelated changes.

Bad: “Fix various SM3 review comments”

Good: “Remove unused MBEDTLS_ERR_SM3_HW_ACCEL_FAILED”, etc. — but each in its own commit!